content/articles/django-celery-tasks/index.md

```diff
@@ -0,0 +1,449 @@
+# set work directory
+WORKDIR /usr/src/app
+
+# set environment variables
+ENV PYTHONDONTWRITEBYTECODE 1
+ENV PYTHONUNBUFFERED 1
+
+# install dependencies
+RUN pip install --upgrade pip
+COPY ./requirements.txt /usr/src/app/requirements.txt
+RUN pip install -r requirements.txt
+
+# copy project
+COPY . /usr/src/app/
+
+RUN python manage.py migrate
+
+EXPOSE 8000
+
+CMD ["python", "manage.py", "runserver", "0.0.0.0:8000"]
+```
+The code snippet above contains directives that will be used to create the Docker image. For more information on how to create Django Docker images, visit [creating Django Docker images](https://www.section.io/engineering-education/django-docker/).
+
+#### Create a Docker compose deployment file
+In the working directory `project`, create a file named `docker-compose.yml` and add the code snippet below.
+```yml
+version: '3.3'
+
+services:
+  web:
+    build: ./celerytask #path to the root project folder
+    command: python manage.py runserver 0.0.0.0:8000
+    volumes:
+      - ./celerytask:/usr/src/app/
+    ports:
+      - 8000:8000 # sets the port that maps to internal port in docker container
+    environment:
+      - DEBUG=1
+      - SECRET_KEY=dbaa1_i7%*3r9-=z-+_mz4r-!qeed@(-a_r(g@k8jo8y3r27%m
+      - DJANGO_ALLOWED_HOSTS=localhost 127.0.0.1 [::1]
+      - CELERY_BROKER=redis://redis:6379/0
+      - CELERY_BACKEND=redis://redis:6379/0
+    depends_on:
+      - redis
+
+  celery:
+    build: ./celerytask
+    command: celery worker --app=core --loglevel=info --logfile=logs/celery.log # Command used to start the Celery worker in the Docker container
+    volumes:
+      - ./celerytask:/usr/src/app
+    environment:
+      - DEBUG=1
+      - SECRET_KEY=dbaa1_i7%*3r9-=z-+_mz4r-!qeed@(-a_r(g@k8jo8y3r27%m
+      - DJANGO_ALLOWED_HOSTS=localhost 127.0.0.1 [::1]
+      - CELERY_BROKER=redis://redis:6379/0
+      - CELERY_BACKEND=redis://redis:6379/0
+      # depends on show that celery worker service requires the web service and the redis service to run
+    depends_on: 
+      - web
+      - redis
+
+  redis:
+    image: redis:6-alpine
+```
+
+In the above `docker-compose.yml` file, we are having 3 services:
+  - `web` - is the service that runs our application code.
+  - `celery`- is the service that runs the Celery worker.
+  - `redis` - is the service that runs the Redis server. Redis is a key-pair datastore that will be used to store the queued events.
+For more information and a getting started guide on Docker compose, visit [Docker compose guide](https://www.section.io/engineering-education/set-up-containerize-and-test-a-single-hadoop-cluster-using-docker-and-docker-compose/)
+
+#### Building the image
+In the root project directory, execute the command below to create an image.
+```bash
+docker build -t celerytask .
+```
+- `-t` adds a tag `celerytask` to the image that will be created.
+- `.` at the end of the command indicates that Dockerfile is in the current directory from where the command is being executed.
```

Comment:
You need a requirements.txt file for the docker image to be built. Please add it.